Build failure: ecm

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```console
nix build -f . ecm
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/305570978

### Relevant log output

```console
> main.c: In function 'main':
       > main.c:902:11: error: implicit declaration of function 'gethostname' [-Wimplicit-function-declaration]
       >   902 |       if (gethostname (mname, MNAMESIZE) == 0)
       >       |           ^~~~~~~~~~~
```

### Additional context

Probably due to GCC 15

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.7, NixOS, 25.11 (Xantusia), 25.11.20250723.fc02ee7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/8797hshpb8qqgfqnyqm8af2rd17lhrva-source`

### Notify maintainers

@roconnor

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ec/ecm/package.nix
+++ b/pkgs/by-name/ec/ecm/package.nix
@@ -26,8 +26,13 @@ stdenv.mkDerivation {
     substituteInPlace test.ecm --replace /bin/rm rm
   '';
 
-  # See https://trac.sagemath.org/ticket/19233
-  configureFlags = lib.optional stdenv.hostPlatform.isDarwin "--disable-asm-redc";
+  configureFlags = [
+    # Otherwise, undesired flags from gmp (such as -std=c99) are leaking
+    "-enable-gmp-cflags=false"
+  ]
+  ++
+    # See https://trac.sagemath.org/ticket/19233
+    lib.optional stdenv.hostPlatform.isDarwin "--disable-asm-redc";
 
   buildInputs = [
     m4

